tests/next_tests/integration_tests/feature_tests/ffront_tests/test_as_offset.py

```diff
@@ -0,0 +1,213 @@
+    def testee(
+        a: gtx.Field[[IDim, KDim], int], offset_field: gtx.Field[[IDim, KDim], int]
+    ) -> gtx.Field[[IDim, KDim], int]:
+        return a(as_offset(Ioff, offset_field))
+
+    out = cases.allocate(cartesian_case, testee, cases.RETURN)()
+    a = cases.allocate(cartesian_case, testee, "a")()
+    offset_field = cases.allocate(cartesian_case, testee, "offset_field").strategy(
+        cases.ConstInitializer(3)
+    )()
+    ref = a[3:]
+
+    cases.verify(
+        cartesian_case,
+        testee,
+        a,
+        offset_field,
+        out=out[: len(out.asnumpy()) - 3],
+        offset_provider={"Ioff": IDim},
+        ref=ref,
+        comparison=lambda out, ref: np.all(out == ref),
+    )
+
+
+@pytest.mark.uses_dynamic_offsets
+def test_offset_field_domain(cartesian_case):
+    i_size = cartesian_case.default_sizes[IDim]
+    k_size = cartesian_case.default_sizes[KDim]
+
+    @gtx.field_operator
+    def testee_fo(
+        a: gtx.Field[[IDim, KDim], int], offset_field: gtx.Field[[IDim, KDim], int]
+    ) -> gtx.Field[[IDim, KDim], int]:
+        return a(as_offset(Ioff, offset_field))
+
+    @gtx.program
+    def testee(
+        a: gtx.Field[[IDim, KDim], int],
+        offset_field: gtx.Field[[IDim, KDim], int],
+        out: gtx.Field[[IDim, KDim], int],
+    ):
+        testee_fo(a, offset_field, out=out, domain={IDim: (1, 5), KDim: (0, 10)})
+
+    out = cases.allocate(cartesian_case, testee, "out")()
+    ref = gtx.as_field([IDim, KDim], np.zeros([i_size, k_size], dtype=int))
+    a = cases.allocate(cartesian_case, testee, "a").extend({IDim: (0, 3)})()
+    offset_field = cases.allocate(cartesian_case, testee, "offset_field").strategy(
+        cases.ConstInitializer(3)
+    )()
+    a_shift = a[3:]
+    ref[1:5] = a_shift.asnumpy()[1:5]
+
+    cases.verify(
+        cartesian_case,
+        testee,
+        a,
+        offset_field,
+        out,
+        inout=out,
+        offset_provider={"Ioff": IDim},
+        ref=ref,
+        comparison=lambda out, ref: np.all(out == ref),
+    )
+
+
+@pytest.mark.uses_dynamic_offsets
+def test_offset_field_input_domain(cartesian_case):
+    i_size = cartesian_case.default_sizes[IDim]
+    k_size = cartesian_case.default_sizes[KDim]
+
+    @gtx.field_operator
+    def testee_fo(
+        a: gtx.Field[[IDim, KDim], int], offset_field: gtx.Field[[IDim, KDim], int]
+    ) -> gtx.Field[[IDim, KDim], int]:
+        return a(as_offset(Ioff, offset_field))
+
+    @gtx.program
+    def testee(
+        a: gtx.Field[[IDim, KDim], int],
+        offset_field: gtx.Field[[IDim, KDim], int],
+        out: gtx.Field[[IDim, KDim], int],
+    ):
+        testee_fo(a, offset_field, out=out)
+
+    a_fo = cases.allocate(cartesian_case, testee, "a").extend({IDim: (0, 3)})().ndarray
+    out = gtx.as_field(
+        [IDim, KDim],
+        np.zeros([i_size, k_size], dtype=int),
+        origin={IDim: 2, KDim: 0},
+    )
+    a = gtx.as_field([IDim, KDim], a_fo, origin={IDim: 2, KDim: 0})
```

Comment:
```suggestion
    out = cartesian_case.as_field(
        [IDim, KDim],
        np.zeros([i_size, k_size], dtype=int),
        origin={IDim: 2, KDim: 0},
    )
    a = cartesian_case.as_field([IDim, KDim], a_fo, origin={IDim: 2, KDim: 0})
```
I guess this is the error.